### PR TITLE
fix: [embed-react] types fix for compatible with TypeScript's "moduleResolution": "bundler"

### PR DESCRIPTION
**Pull request by LarsArtmann**
# fix: [embed-react] types fix for compatible with TypeScript's "moduleResolution": "bundler"
- moved types in exports to the first position to comply with the standard 
- updated version so a new release can be rolled out
# References 
#11432 
#11408

### CODE REVIEW

File: packages/embeds/embed-react/package.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
     "withEmbedPublishEnv": "NEXT_PUBLIC_EMBED_LIB_URL='https://app.cal.com/embed/embed.js' NEXT_PUBLIC_WEBAPP_URL='https://app.cal.com' yarn",
     "prepack": "yarn ../../../ lint --filter='@calcom/embed-react' && yarn withEmbedPublishEnv build && yarn packaged:tests",
     "embed-web-start": "yarn workspace @calcom/web start",
-    "embed-dev": "yarn workspace @calcom/embed-react dev",
+    "embed-dev": "yarn workspace @calcoterm/embed-react dev",
```
Comment (PeerRich) [inline packages/embeds/embed-react/package.json:26]: i doubt this should be calcoterm?

Comment (PeerRich) [inline packages/embeds/embed-react/package.json:26]:
```suggestion
    "embed-dev": "yarn workspace @calcom/embed-react dev",
```

**Review by PeerRich — APPROVED** [review]

File: packages/embeds/embed-react/package.json (hunk 1 above)
Comment (LarsArtmann) [reply packages/embeds/embed-react/package.json]: Seems like a reasonable add-on to this fix.

**Comment by lottamus** [comment]
@PeerRich can you publish the `@calcom/embed-react` package that includes this fix for typings? (v1.3.1) 🙏 

https://www.npmjs.com/package/@calcom/embed-react?activeTab=versions

image

**Comment by Larkify** [comment]
Is this ever going to be published? It's been months and this is a breaking change that prevents anyone trying to use this package from being able to compile it

**Comment by LarsArtmann** [comment]
Since this is still not published, I wanted to share my current work around.
It was created just for me. I'm using pnpm. Feel free to change it to fix your project.

## Adding it to your scripts in the package.json will let it run after every install.
```json
...
"scripts": {
  "postinstall": "node fix-type.js",
}
...
```

### TypeScript version
```ts
import fs from "node:fs"

function fixType() {
	fixTypeAt("./node_modules/@calcom/embed-react/package.json")
	fixTypeAt("../../node_modules/@calcom/embed-react/package.json")
}

function fixTypeAt(path: string) {
	if (!fs.existsSync(path)) {
		console.warn("Cannot find package.json at", path)
		return
	}
	console.log("Found package.json at", path)

	const packageJsonRaw = fs.readFileSync(path, "utf8")
	const packageJson = JSON.parse(packageJsonRaw)

	packageJson.exports["."].types = "./dist/embed-react/src/index.d.ts"

	fs.writeFileSync(path, JSON.stringify(packageJson, null, 2))
}

fixType()
```

### JavaScript version
```js
import fs from "node:fs";
function fixType() {
    fixTypeAt("./node_modules/@calcom/embed-react/package.json");
    fixTypeAt("../../node_modules/@calcom/embed-react/package.json");
}
function fixTypeAt(path) {
    if (!fs.existsSync(path)) {
        console.warn("Cannot find package.json at", path);
        return;
    }
    console.log("Found package.json at", path);
    const packageJsonRaw = fs.readFileSync(path, "utf8");
    const packageJson = JSON.parse(packageJsonRaw);
    packageJson.exports["."].types = "./dist/embed-react/src/index.d.ts";
    fs.writeFileSync(path, JSON.stringify(packageJson, null, 2));
}
fixType();
```

**Pull request merged**